File: programs/src/echo.c
#include <stdio.h>
#include <sys/raw.h>

int main(size_t argc, char **argv) {
  if (argc == 1) {
    printf("need at least one argument to echo!\n");
    return -1;
  }

  for (int i = 1; i < argc; i++)
    printf("%s", argv[i]);

  printf("\n");

  return 0;
}

File: bin/src/echo.zig
const libc = @import("libc");
const printf = libc.stdio.zprintf;
const Error = libc.sys.errno.Error;

pub fn main() Error!void {
    var args = libc.sys.args();
    if (args.count() < 2) {
        try printf("expected at least one argument to echo...\n", .{});
        return error.NotEnoughArguments;
    }

    _ = args.next();
    while (args.next()) |arg| {
        try printf("%s", .{arg.ptr});
    }
    try printf("\n", .{});
}

comptime {
    _ = libc;
}
